scripts/cleanup-all.py

```diff
@@ -16,6 +17,30 @@ def delete_everything(pc):
     for index in pc.list_indexes().names():
         try:
             print("Deleting index: " + index)
+            desc = pc.describe_index(index)
+
+            # Check whether index can be deleted
+            if desc.deletion_protection == "enabled":
+                pc.configure_index(index, deletion_protection="disabled")
+
+            # Wait for index to be ready before deleting
+            ready_to_delete = False
+            max_wait = 60
+            time_waited = 0
+            while not ready_to_delete:
+                desc = pc.describe_index(index)
+                if desc.status.state == "Ready":
+                    ready_to_delete = True
+                    break
+                else:
+                    print("Index is not ready yet. Waiting for 2 seconds.")
+                    time.sleep(2)
+                    time_waited += 2
+
+                if time_waited > max_wait:
```

Comment:
small nit: because of logic ordering, you end up sleeping for 2 seconds before failing on the last iteration.

Reply:
This is what comes from outsourcing too much typing to github copilot 🤣